node/heartbeat/monitor.go

```diff
@@ -66,7 +66,7 @@ func NewMonitor(
 				return nil, err
 			}
 
-			mhbi.shardID = shardId
+			mhbi.computedShardID = shardId
```

Comment:
Please add here a TODO as to not forget to recompute "computedShardID" field each time we receive a new message from the peer. Or even better, add a call to a func on line 124 and that func will have this TODO

Reply:
computedShardID has to be updated from peer processor / node coordinator, it will be updated at the end of the epoch / start of the new.

Reply:
Might add a value which is treated as unallocated. so peer if not allocated to any shard.

Reply:
done. made a function which has that TODO